packages/loopback/lib/context/inject.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import 'reflect-metadata';
+
+const REFLECTION_KEY = 'loopback.inject';
+
+/**
+ * A decorator to annotate method arguments for automatic injection
+ * by LoopBack IoC container.
+ *
+ * Usage - Typescript:
+ *
+ * ```ts
+ * class InfoController {
+ *   constructor(@inject('application.name') public appName: string) {
+ *   }
+ *   // ...
+ * }
+ * ```
+ *
+ * Usage - JavaScript:
+ *
+ *  - TODO(bajtos)
+ *
+ * @param bindingKey What binding to use in order to resolve the value
+ * of the annotated argument.
+ */
+export function inject(bindingKey: string) {
```

Comment:
Let's leave that out of scope of this initial work please, see https://github.com/strongloop/loopback-next/pull/176#discussion_r112399332